pwa/src/templates/componentDetail/ComponentsDetailTemplate.tsx

```diff
@@ -195,13 +200,87 @@ export const ComponentsDetailTemplate: React.FC<ComponentsDetailTemplateProps> =
             <InfoCard
               title=""
               content={
-                <RatingIndicatorTemplate
-                  layoutClassName={styles.ratingIndicatorContainer}
-                  component={_getComponent.data}
-                />
+                <>
+                  {_getComponent.data.embedded?.rating ? (
+                    <RatingIndicatorTemplate
+                      layoutClassName={styles.ratingIndicatorContainer}
+                      maxRating={_getComponent.data.embedded?.rating?.maxRating}
+                      rating={_getComponent.data.embedded?.rating?.rating}
+                    />
+                  ) : (
+                    <div className={styles.noRatingStyle}>{t("No rating available")}</div>
+                  )}
+                  <span
+                    onClick={() => {
+                      show();
+                    }}
+                    className={styles.link}
+                  >
+                    <Link icon={<ArrowRightIcon />} iconAlign="start">
+                      Rating
+                    </Link>
+                  </span>
+                </>
               }
               layoutClassName={styles.infoCard}
             />
+            {isVisible && (
+              <div className={styles.overlay}>
+                <NotificationPopUp
+                  {...{ hide, isVisible }}
+                  title={"Rating"}
+                  description={
```

Comment:
Right now, we're building a RatingTemplate within the ComponentDetailTemplate. This file is now 580 lines, which is an indicator we should be separating some logic and templates.

I suggest creating a template part which holds this rating overview. And just adding this template part within this description.